### ISSUE TITLE
Color in output

### ISSUE THREAD

**Issue by RomeoV** [tninja/aider.el#9]
Awesome package, thanks. Any chance we can enable color in the aider output inside emacs aswell? Currently it's a bit harder to read the code snippets. Presumably we'd need to forward the color codes that aider already outputs correctly to emacs?

**Comment by tninja** [comment]
Yeah agree that the current comint based solution lost the code color from
aider output. i do try ansi color based change for the coming aider session
buffer, suggested by ChatGPT, doesn’t work. ChatGPT said aider might used
some more advanced coloring way.

**Comment by CeleritasCelery** [comment]
Aider does not send the color codes to the shell because we are telling it this a `"dumb"` terminal via `comint-terminfo-terminal`. If you set it to something else like `"eterm-color"` (an Emacs terminal) or `"xterm-color"` it will colorize the output, but it also screws things up because it sends a lot of other commands that `comint` doesn't understand and also expects to get to receive "cursor position requests" from Emacs (which it doesn't send). I am not sure if there is a value that gives just color output with breaking other things. Aider uses [prompt-toolkit](https://github.com/prompt-toolkit/python-prompt-toolkit) to handle the coloring and it checks for a dumb terminal [here](https://github.com/prompt-toolkit/python-prompt-toolkit/blob/master/src/prompt_toolkit/utils.py#L317). See also [prompt-toolkit/python-prompt-toolkit/#1032](https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1032). Unfortunately that setting both disables interactive editing *and* colors, so I don't see a way to only enable colors.

**Comment by tninja** [comment]
Trying to use the major mode of buffer which start aider, to highlight the comint aider buffer session. Tested elisp and python and it seems working

**Comment by tninja** [comment]
merged in https://github.com/tninja/aider.el/pull/21

**Comment by katafrakt** [comment]
Indeed it does work for elisp, but for some reason it does not for Elixir for me. I'm using `elixir-ts-mode` and `prog-mode` is its parent. I wonder if it's maybe because of tree-sitter somehow.

**Comment by tninja** [comment]
I asked LLM for aider.el, using this question (with aider.el file added): inside aider-run-aider, i use aider--inherit-source-highlight, to get code highlight from current major-mode. however, it doesn't work for some particular major mode, eg. elixir-ts-mode. Do you know why?

It do give me some example code. part of it looks like:
```
        (if source-tree-sitter
            (progn
              (require 'tree-sitter)
              (require 'tree-sitter-langs)
              (tree-sitter-mode)
              (tree-sitter-hl-mode))
```

But I am not familiar with elixir-ts-mode. wonder if you want to take a try and see if it works?

**Comment by katafrakt** [comment]
Thanks for the reply. I tried this and tried to iterate on top of that, but only got strings to be colored. I will try some more if I have time and some other ideas to try.

**Comment by CeleritasCelery** [comment]
I don’t use `elixir-ts-mode` but if I had to guess it is because tree sitter parsers tend to be quite strict. Does it work better using the non-tree sister mode?

**Comment by tninja** [comment]
I just noticed the output format of aider, is basically just markdown.. and mardown-mode can render both text and code block with (setq markdown-fontify-code-blocks-natively t), Actually right now gptel output buffer is just using markdown mode.

If we can figure out how can we apply that on current aider-comint-mode, that might be able to solve the problem with relative less code change.

**Comment by chuxubank** [comment]
How about [mistty](https://github.com/szermatt/mistty)?

**Comment by tninja** [comment]
Thanks for the reference. Is it a new type of vterm? It is under active
development, which is good. I wish it can be more battle tested, and no
hard requirements for emacs 29.1. Lots of corporate Linux distro is
relative old and doesn’t support 29.1.

**Comment by tninja** [comment]
Closing this since aider.el now use markdown-mode.el to render the color.
